File: BackEnd/XeGo-Microservices/Services/Auth/XeGo.Services.Auth.API/Models/Dto/RefreshTokenRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace XeGo.Services.Auth.API.Models.Dto
{
    public class RefreshTokenRequestDto
    {
        [Required] public string RefreshToken = string.Empty!;
        [Required] public string UserId = string.Empty!;
        [Required] public string FromApp = string.Empty!;
    }
}

File: FrontEnd/xego_rider/lib/models/Dto/refresh_token_request_dto.dart
class RefreshTokenRequestDto {
  RefreshTokenRequestDto({
    required this.refreshToken,
    required this.userId,
    required this.fromApp,
  });

  final String refreshToken;
  final String userId;
  final String fromApp;

  factory RefreshTokenRequestDto.fromJson(Map<String, dynamic> json) {
    return RefreshTokenRequestDto(
      refreshToken: json['refreshToken'],
      userId: json['userId'],
      fromApp: json['fromApp'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'refreshToken': refreshToken,
      'userId': userId,
      'fromApp': fromApp,
    };
  }
}
